### DISCUSSION TITLE
Headless x-combobox multi select remove undefined

### DISCUSSION THREAD

**Discussion by alfmatos** [alpinejs/alpine#3785] [7. Bugs (archived)]
There's an issue that can be replicated on the x-combobox sample page, and that we are hitting in our own code as well.

When adding items by order, and removing the last one on the pill directly triggers an inconsistent state.
Removing the last entry, removes every selected option, and then we get an undefined error.

**Steps to reproduce**

On the multiselect combobox on the [headless combobox page](https://alpinejs.dev/component/headless-combobox):

- Press 'a' to search and see selectable frameworks
- Select, "Laravel", "Ruby on Rails" and "Django"
- Remove "Django" through the pill box above the input
- All items are deselected on the dropdown (check mark disappears, confirmed by pressing 'a' again to see framework)
- Pill items remain visible
- Select another element in the combobox triggers an undefined

```
Uncaught TypeError: Cannot read properties of undefined (reading 'after')
    at cdn.min.js:5:33020
    at h (cdn.min.js:1:3258)
    at cdn.min.js:5:32953
    at Ne (cdn.min.js:5:1361)
    at cdn.min.js:5:1043
    at cr (cdn.min.js:1:6458)
    at Bi (cdn.min.js:5:32050)
    at cdn.min.js:5:31864
    at Array.r (cdn.min.js:5:16159)
    at En (cdn.min.js:1:270)
(anonymous) @ cdn.min.js:5
h @ cdn.min.js:1
(anonymous) @ cdn.min.js:5
Ne @ cdn.min.js:5
(anonymous) @ cdn.min.js:5
cr @ cdn.min.js:1
Bi @ cdn.min.js:5
(anonymous) @ cdn.min.js:5
r @ cdn.min.js:5
En @ cdn.min.js:1
setTimeout (async)
ee @ cdn.min.js:5
cr @ cdn.min.js:1
Bi @ cdn.min.js:5
(anonymous) @ cdn.min.js:5
r @ cdn.min.js:5
En @ cdn.min.js:1
```

**Comment by alfmatos** [comment]
Tried to create an Issue directly, but it's redirecting to discussions. Created #3786 from this discussion.

**Reply by ekwoka** [reply]
Issues aren't allowed on this repo. Only discussions
